// reactstrap-alert/build/es/PromptModal.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React, { useState } from 'react';
import { Button, Input, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
export function PromptModal(props) {
  var _useState = useState(true),
      isOpen = _useState[0],
      _toggle = _useState[1];

  var _useState2 = useState(''),
      value = _useState2[0],
      setValue = _useState2[1];

  function toggle() {
    _toggle(false);

    props.resolve(false);
  }

  function onResolve() {
    _toggle(false);

    props.resolve(value);
  }

  return React.createElement(Modal, _extends({
    isOpen: isOpen,
    toggle: toggle
  }, props.modalProps || {}), React.createElement(ModalHeader, _extends({
    toggle: toggle
  }, props.modalHeaderProps || {}), props.title), React.createElement(ModalBody, props.modalBodyProps || {}, props.message, React.createElement(Input, _extends({
    value: value,
    onChange: function onChange(e) {
      return setValue(e.target.value);
    }
  }, props.inputProps || {}))), React.createElement(ModalFooter, props.modalFooterProps || {}, React.createElement(Button, _extends({
    disabled: !value,
    className: props.confirmColor || 'primary',
    onClick: onResolve
  }, props.confirmButtonProps || {}), props.confirmText || 'Продолжить')));
}
